### PR TITLE
test(manager 3.0): Added test for on-resume-start-tasks flag

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fgelcer — CHANGES_REQUESTED** [review]
please add description in the commit message about the feature, and what this test is aiming to cover (don't need to explain what the code already does), but empty commit messages is a very bad practice.

**Review by fgelcer — APPROVED** [review]
with this descriptive commit message, now i can understand what the test is trying to do...
LGTM

**Review by rayakurl — APPROVED** [review]
Approved

**Comment by fgelcer** [comment]
@ShlomiBalalis , why did you re-work it, after it was already approved?

**Comment by ShlomiBalalis** [comment]
> @ShlomiBalalis , why did you re-work it, after it was already approved?

The test failed because the flag wasn't properly explained (no one even mentioned the duration flag) so I rework the test to work accordingly and created an opposite test that does not take the duration into account and expects a different result. You can read further in the commit message

File: sdcm/mgmt/cli.py [hunk 1]
```diff
@@ -806,8 +810,8 @@ def resume(self, start_tasks=True):
         self.sctool.run(cmd=cmd)
 
     @contextmanager
-    def suspend_manager_then_resume(self, start_tasks=True):
-        self.suspend()
+    def suspend_manager_then_resume(self, start_tasks=True, start_tasks_in_advance=False, duration=None):
```
Comment (fgelcer) [inline sdcm/mgmt/cli.py:813]: what is this duration about? is it the time it will be stopped? or is it some kind of delay until restarted?

Comment (ShlomiBalalis) [reply sdcm/mgmt/cli.py:813]: It's the time that if the cluster is resumed before it passes, the tasks will not be continued on resume

Comment (fgelcer) [reply sdcm/mgmt/cli.py:813]: perhaps a better naming for `duration` is required

Comment (ShlomiBalalis) [reply sdcm/mgmt/cli.py:813]: I understand why you'd like this change, but I'd rather keep the function arguments name the same as the name given to them in the cli

**Review by fgelcer — APPROVED** [review]
> I understand why you'd like this change, but I'd rather keep the function arguments name the same as the name given to them in the cli

open an issue to the manager, as the naming is very bad (at least, from this POV)

File: mgmt_cli_test.py [hunk 2]
```diff
@@ -1012,6 +1016,42 @@ def _suspend_and_resume_task_template(self, task_type):
...
+            assert suspendable_task.wait_for_status(list_status=[TaskStatus.DONE], timeout=1200, step=10), \
+                f"After the cluster was resumed (while resuming AFTER the suspend duration has passed)," \
+                f" task {suspendable_task.id} failed to reach status " \
+                f"{TaskStatus.DONE}, but instead stayed in {suspendable_task.status}"
+        else:
+            try:
+                suspendable_task.wait_for_status(list_status=TaskStatus.all_statuses() - {TaskStatus.STOPPED},
+                                                 timeout=1200, step=10)
+            except RetryError:
+                pass
```
Comment (rayakurl) [inline mgmt_cli_test.py:1048]: I would write here to the log that there was an error like expected

**Comment by fruch** [comment]
@ShlomiBalalis is this still needed ? or can be closed ?

**Review by rayakurl — COMMENTED** [review]
Looks good. but since it's a pretty old PR, please run it again to make sure that it works

**Comment by ShlomiBalalis** [comment]
> Looks good. but since it's a pretty old PR, please run it again to make sure that it works

Passed! - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/Shlomo/job/centos-sanity-test/20/

**Review by rayakurl — APPROVED** [review]
approved

**Review by fgelcer — APPROVED** [review]
LGTM

**Pull request merged**